repos/system_upgrade/el7toel8/actors/authselectscanner/actor.py

```diff
@@ -0,0 +1,222 @@
+
+        if self.pam.has('pam_winbind'):
+            profile = 'winbind'
+            enabled_no += 1
+
+        if is_service_enabled('ypbind'):
+            profile = 'nis'
+            enabled_no += 1
+
+        self.profile = profile if enabled_no == 1 else None
+
+    def step_detect_features(self):
+        pam_map = {
+            'pam_faillock': 'with-faillock',
+            'pam_fprintd': 'with-fingerprint',
+            'pam_access': 'with-pamaccess',
+            'pam_mkhomedir': 'with-mkhomedir',
+            'pam_oddjob_mkhomedir': 'with-mkhomedir'
+        }
+
+        for module, feature in pam_map.items():
+            if self.pam.has(module):
+                self.features.append(feature)
+
+    def step_detect_sssd_features(self):
+        if self.profile != "sssd":
+            return
+
+        # sudoers: sss
+        result = re.search(
+            "^[ \t]*sudoers[ \t]*:.*sss.*$",
+            read_file("/etc/nsswitch.conf"),
+            re.MULTILINE
+        )
+
+        if result is not None:
+            self.features.append("with-sudo")
+
+        # SSSD Smartcard support
+        # We enable smartcard support only if it was not handled by pam_pkcs11.
+        # Otherwise pam_pkcs11 configuration must be converted manually.
+        if not self.pam.has('pam_pkcs11'):
+            if self.ac.get_bool('USESMARTCARD'):
+                self.features.append("with-smartcard")
+
+            if self.ac.get_bool('FORCESMARTCARD'):
+                self.features.append("with-smartcard-required")
+
+            if self.dconf.get_string('removal-action') == 'lock-screen':
+                self.features.append("with-smartcard-lock-on-removal")
+
+    def step_detect_winbind_features(self):
+        if self.profile != "winbind":
+            return
+
+        if self.ac.get_bool('WINBINDKRB5'):
+            self.features.append("with-krb5")
+
+    def step_detect_if_confirmation_is_required(self):
+        sysconfig = '/etc/sysconfig/authconfg'
+        links = {
+            '/etc/pam.d/fingerprint-auth': '/etc/pam.d/fingerprint-auth-ac',
+            '/etc/pam.d/password-auth': '/etc/pam.d/password-auth-ac',
+            '/etc/pam.d/postlogin': '/etc/pam.d/postlogin-ac',
+            '/etc/pam.d/smartcard-auth': '/etc/pam.d/smartcard-auth-ac',
+            '/etc/pam.d/system-auth': '/etc/pam.d/system-auth-ac'
+        }
+
+        self.confirm = True
+
+        # Check that authconfig was used to create the configuration
+        if not os.path.isfile(sysconfig):
+            return
+
+        # Check that all files are symbolic links to authconfig files
+        for name, target in links.items():
+            if not os.path.islink(name):
+                return
+
+            if os.readlink(name) != target:
+                return
+
+        # Check that all file were not modified after
+        # /etc/sysconfig/authconfig was created.
+        mtime = os.path.getmtime(sysconfig)
+        for file in links.values():
```

Comment:
overridden python file built-in at L218, rename?

Reply:
Renamed to `f`.